src/Workspaces/Core/Portable/Storage/SQLite/v1/Interop/SafeSqliteChildHandle`1.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.SQLite.v1.Interop
+{
+    /// <summary>
+    /// The base handle type for an SQLite resource that exists within the context of a parent handle, and should always
+    /// be released prior to the parent handle.
+    /// </summary>
+    /// <typeparam name="T">The SQLite resource wrapper type.</typeparam>
+    internal abstract class SafeSqliteChildHandle<T> : SafeSqliteHandle<T>
+        where T : class
```

Comment:
thanks.  this is what i was thinking about, and i like how it does clean things up.